prophic3/prophic3.h

```diff
@@ -32,7 +32,10 @@ namespace prophic3
 
     msat_env msat_env_;
     const ic3ia::TransitionSystem & conc_ts_;
-    Rewriter rw_;
```

Comment:
can we handle multi-dimensional arrays without rewriting?

Reply:
Hmm... that's a good question. I'll think about that, and I could write a simple test.

Reply:
All right, so I think it's fine. It turns out there's already a test in `prophic3/samples/` for 2d arrays and then I added a couple more. It was able to prove the true one and catch the bug in the false one so it appears to be all right. I also checked the witness and it passed. This makes sense to me, because this simplified axiom enumeration is just adding array axioms exactly as an array solver would. Thanks for thinking to test this!